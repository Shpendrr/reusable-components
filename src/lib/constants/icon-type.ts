export enum IconType {
    DOWN_ARROW_ICON = 'down-arrow-icon',
}