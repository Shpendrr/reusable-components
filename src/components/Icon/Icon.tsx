import { IconType } from "@/lib/constants/icon-type";
import { memo } from "react";
import DownArrowIcon from "@/components/Icon/IconType/DownArrowIcon";

interface IIconProps {
  iconType: IconType | string;
  className?: string;
}

interface IconComponentMap {
  [key: string]: React.MemoExoticComponent<
    ({ className }: { className?: string }) => JSX.Element
  >;
}

const iconComponents: IconComponentMap = {
  [IconType.DOWN_ARROW_ICON]: DownArrowIcon,
};

function Icon({ iconType, className }: IIconProps) {
  const IconComponent = iconComponents[iconType];
  if (!IconComponent) {
    return null;
  }
  return <IconComponent className={className} />;
}

export default memo(Icon);
