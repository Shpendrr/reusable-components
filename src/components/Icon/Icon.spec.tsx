import { render } from "@testing-library/react";
import { IconType } from "@/lib/constants/icon-type";
import "@testing-library/jest-dom";
import Icon from "@/components/Icon/Icon";

describe("Icon", () => {
  it("renders the DownArrowIcon component when iconType is IconType.DOWN_ARROW_ICON", () => {
    const { container } = render(<Icon iconType={IconType.DOWN_ARROW_ICON} />);
    expect(container.querySelector(".down-arrow-icon")).toBeInTheDocument();
  });

  it("does not render any component when iconType is not recognized", () => {
    const { container } = render(<Icon iconType="unknown-icon" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
