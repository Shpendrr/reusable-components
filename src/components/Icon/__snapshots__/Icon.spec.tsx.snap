// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon does not render any component when iconType is not recognized 1`] = `null`;
